com.reprezen.swagedit/src/com/reprezen/swagedit/assist/SwaggerContentAssistProcessor.java

```diff
@@ -48,15 +51,36 @@
  * This class provides basic content assist based on keywords used by the
  * swagger schema.
  */
-public class SwaggerContentAssistProcessor extends TemplateCompletionProcessor implements IContentAssistProcessor {
+public class SwaggerContentAssistProcessor extends TemplateCompletionProcessor implements IContentAssistProcessor, ICompletionListener {
 
 	private final SwaggerProposalProvider proposalProvider = new SwaggerProposalProvider();
+	private final JsonReferenceProposalProvider referenceProposalProvider = new JsonReferenceProposalProvider();
+	private final ContentAssistant contentAssistant;
+
 	private String currentPath = null;
+	private int cyclePosition = 0;
+	private boolean isRefCompletion = false;
+
+	private String[] textMessages = new String[] {
+			"Press 'Ctrl+Space' to show all schemas in the project.",
+			"Press 'Ctrl+Space' to show all schemas in the workspace.",
+			"Press 'Ctrl+Space' to show schemas in the current file."
+	};
 
-	public SwaggerContentAssistProcessor() {}
+	public SwaggerContentAssistProcessor() {
+		this(null);
+	}
+
+	public SwaggerContentAssistProcessor(ContentAssistant ca) {
+		this.contentAssistant = ca;
+	}
 
 	@Override
 	public ICompletionProposal[] computeCompletionProposals(ITextViewer viewer, int documentOffset) {
+		if (isRefCompletion) {			
+			cyclePosition = cyclePosition == 2 ? 0 : cyclePosition + 1;
```

Comment:
`2` is a magic value here. We could use something like `cyclePosition = (cyclePosition + 1) % textMessages.length;`